microprofile/WFLY-14940_upgrade-microprofile-health-3.1.adoc

```diff
@@ -0,0 +1,69 @@
+= Upgrade MicroProfile Health to 3.1
+:author:            Martin Stefanko
+:email:             mstefank@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          microprofile,health,observability
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+
+MicroProfile 4.1 provides a minor upgrade for https://github.com/eclipse/microprofile-health/releases/tag/3.1[MicroProfile Health 3.1] with changes described in its https://download.eclipse.org/microprofile/microprofile-health-3.1/microprofile-health-spec-3.1.html#release_notes_3_1[Release Notes].
+
+This component upgrade also covers the upgrade of smallrye-health 3.1.1 that implements MicroProfile Health 3.1.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-14940[WFLY-14040]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1763[EAP7-1763]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+=== Testing By
+
+[X] Engineering
+[ ] QE
+
+== Requirements
+
+* Upgrade MicroProfile Health to 3.1
+* Upgrade SmallRye Health to 3.1.1
+* Pass all MicroProfile Health tests in WildFly integration test suite
+* Pass the MicroProfile Health 3.1 TCK without failures
+* Adjust the tests to work as needed
+* Adjust QuickStarts to showcase new Startup feature
+
+
+=== Non-Requirements
+
+== Implementation Plan
```

Comment:
The [proposal template](https://github.com/wildfly/wildfly-proposals/blob/main/design-doc-template.adoc) has no longer an `Implementation Plan` section bit I find that this can be useful for readers and history.
So maybe this can be left here by adding the items which are now in the `Requirements` one, if we agree in my reasoning in the comment I added there.
Or this can be added to the `Overwiew` section, again, together with the items which are now listed as `Requirements`.

Reply:
Can be resolved (I don't have the grants for doing so), thanks.